tslib/src/functions/school-pin-generator/school-pin-replenishment.data.service.ts

```diff
@@ -0,0 +1,74 @@
+import { SqlService, ISqlParameter } from '../../sql/sql.service'
+import { School, SchoolPinUpdate } from './school-pin-replenishment.service'
+import { TYPES } from 'mssql'
+
+export interface ISchoolPinReplenishmentDataService {
+  getAllSchools (): Promise<School[]>
+  updatePin (schoolPinUpdate: SchoolPinUpdate): Promise<void>
+  getSchoolByUuid (uuid: string): Promise<School | undefined>
+}
+
+export class SchoolPinReplenishmentDataService implements ISchoolPinReplenishmentDataService {
+
+  private sqlService: SqlService
+  constructor () {
+    this.sqlService = new SqlService()
+  }
+
+  async getSchoolByUuid (uuid: string): Promise<School | undefined> {
+    const sql = `
+    SELECT s.id, s.name,  s.pinExpiresAt, s.pin, sce.timezone
+    FROM mtc_admin.school s
+    LEFT OUTER JOIN mtc_admin.sce ON s.id = sce.school_id
+    WHERE s.urlSlug = @schoolUuid AND
+    (s.pinExpiresAt <= GETUTCDATE() OR s.pinExpiresAt IS NULL)`
```

Comment:
I don't understand - the pin is either in date (valid) or is not set (invalid)?

Reply:
select one school, but only if there is no pin, or the expiry date has passed.
am i missing something?

Reply:
I am confused by this query as well. Is that supposed to fetch **only** schools with expired pins or ones that dont have pins at all?

Reply:
I am confused by this query as well. Is that supposed to fetch only schools with expired pins or ones that dont have pins at all? Is that because we want to reassign pins later on to schools that match these criteria?
Hopefully I am getting this right.

Reply:
```
s.pinExpiresAt <= GETUTCDATE()
```
I was misreading this date comparison.  🙈 I agree with you now 😁